### PR TITLE
Porting Fixes to 5.6.x branch

### PR DESCRIPTION
**Pull request by 1akshitha**
### Proposed changes in this pull request

Porting the fixes https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/87 and https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/90 to 5.6.x version to fix https://github.com/wso2/api-manager/issues/452 in APIM manager 4.1.0.

### CODE REVIEW

**Review by ShanChathusanda93 — APPROVED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 1]
```diff
@@ -129,6 +148,24 @@ public boolean canHandle(HttpServletRequest request) {
...
 
+
+    private boolean isInitialRequest(AuthenticationContext context, HttpServletRequest request) {
+
+        return !context.isLogoutRequest() && !hasCodeParamInRequest(request) && !hasErrorParamInRequest(request);
+    }
+
+    private boolean hasErrorParamInRequest(HttpServletRequest request) {
+
+        String error = request.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR);
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:159]: We do not need this variable right?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 2]
```diff
@@ -129,6 +148,24 @@ public boolean canHandle(HttpServletRequest request) {
...
+        return StringUtils.isNotBlank(error);
+    }
+
+    private boolean hasCodeParamInRequest(HttpServletRequest request) {
+
+        String code = request.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE);
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:165]: We do not need this variable right?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java (hunk 1 above)
Comment (1akshitha) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: Fixed with 942f1989b7db43df06a41c8904171bef68e692b6

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java (hunk 2 above)
Comment (1akshitha) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: Fixed with 942f1989b7db43df06a41c8904171bef68e692b6

**Pull request merged**